sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java

```diff
@@ -86,14 +86,26 @@ public OfferEvaluator(
             MesosResourcePool resourcePool = new MesosResourcePool(
                     offer,
                     OfferEvaluationUtils.getRole(podInstanceRequirement.getPodInstance().getPod()));
+
+            Map<TaskSpec, GoalStateOverride> overrideMap = new HashMap<>();
+            for (TaskSpec taskSpec : podInstanceRequirement.getPodInstance().getPod().getTasks()) {
+                GoalStateOverride override =
+                        stateStore.fetchGoalOverrideStatus(
+                                TaskSpec.getInstanceName(podInstanceRequirement.getPodInstance(), taskSpec))
+                                .target;
+
+                overrideMap.put(taskSpec, override);
+            }
+
             PodInfoBuilder podInfoBuilder = new PodInfoBuilder(
                     podInstanceRequirement,
                     serviceName,
                     getTargetConfig(podInstanceRequirement, thisPodTasks.values()),
                     schedulerConfig,
                     thisPodTasks.values(),
                     stateStore.fetchFrameworkId().get(),
-                    useDefaultExecutor);
+                    useDefaultExecutor,
```

Comment:
Since this is a builder, could we not construct it earlier and then call an `addOverride()` method in the task loop above? It seems as if extending the argument list for the construction of a builder goes against the builder pattern a little. (or is this not that kind of Builder?)

Reply:
Not that kind of builder.  As currently structured the constructor needs the override information, so a `.add` style call would be too late.